### ISSUE TITLE
'ExtractInterfaceCodeRefactoringProvider' encountered an error and has been disabled.

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#43952]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1014268/extractinterfacecoderefactoringprovider-encountere-1.html)._

---
Visual Studio 2019 16.4.2
Windows Forms VB.Net project.

When I attempt to extract interface on any windows forms class I get the following:
**'ExtractInterfaceCodeRefactoringProvider' encountered an error and has been disabled.**

Stack trace:

```
System.Collections.Generic.KeyNotFoundException : The given key was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at Microsoft.CodeAnalysis.VisualBasic.ExtractInterface.VisualBasicExtractInterfaceService.CreateFinalSolution(Solution solutionWithInterfaceDocument,IEnumerable`1 documentIds,Dictionary`2 docToRootMap)
   at async Microsoft.CodeAnalysis.VisualBasic.ExtractInterface.VisualBasicExtractInterfaceService.VB$StateMachine_7_UpdateMembersWithExplicitImplementationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.ExtractInterface.AbstractExtractInterfaceService.GetSolutionWithOriginalTypeUpdatedAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.ExtractInterface.AbstractExtractInterfaceService.ExtractInterfaceToNewFileAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.ExtractInterface.AbstractExtractInterfaceService.ExtractInterfaceFromAnalyzedTypeAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.ExtractInterface.ExtractInterfaceCodeAction.ComputeOperationsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeActions.CodeActionWithOptions.GetOperationsAsync(<Unknown Parameters>)
   at Roslyn.Utilities.TaskExtensions.WaitAndGetResult_CanCallOnBackground[T](Task`1 task,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.InvokeWorker(Func`1 getFromDocument,IProgressTracker progressTracker,CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.<>c__DisplayClass20_0. <InvokeCore>b__0()
   at Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformAction(IExtensionManager extensionManager,Object extension,Action action)
```

---
#### Visual Studio Feedback System on 5/3/2020, 01:47 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by allisonchou** [comment]
I have been unable to repro this issue in 16.5.4 and have asked the user for more details in the original ticket.

**Comment by Git2930** [comment]
Hi @allisonchou, 

I have realised that it happens only in VB.Net Windows Forms that contain controls that have had their Modifiers property changed from the default "Friend" value to "Public" in the control's Properties 
window. I am working on a large complex project with many forms that have their controls modifiers set to Public.

I have since updated to 16.5.4 and that seems to have fixed the problem for some forms in my large VB.Net project which is an improvement. But for other forms it continues to happen. So there are some forms that now Extract Interface OK that were previously throwing the exception, but other forms that continue to experience the problem. I don't know why some work and others do not. 

However, I can get the issue to happen in a simple VB.Net project in 16.5.4. See if this recreates the issue for you, it causes the issue for me:
1. Create a new VB.Net Windows Forms Project
2. In Form1.vb, drag a control from the toolbox onto the form, for example a CheckBox
3. In the control's Properties window, change the Modifiers property from the default value "Friend" to "Public"
4. In Form1.vb, place your cursor in the class name
5. Select Edit > Refactor > Extract Interface
6. The KeyNotFoundException happens

I am just confused now why it continues to happen in such a simple example project but no longer occurs for SOME forms in my large complex project after upgrading Visual Studio to 16.5.4
